pkg/registry/apis/provisioning/jobs/syncer.go

```diff
@@ -59,20 +59,20 @@ func NewSyncer(
 func (r *Syncer) Sync(ctx context.Context, complete bool) (string, error) {
 	// FIXME: how to handle the scenario in which folder changes?
 	cfg := r.repository.Config()
+	if !cfg.Spec.Sync.Enabled {
```

Comment:
I think this flag should be in more spots (e.g. why a webhook if sync is not enabled? why triggering a sync job at all?, etc).

I guess we need to add a behaviour in the code on when sync is enabled on create and update to start it.